### PR TITLE
[NUI] Add EncodedImageBuffer class (Load image frome Stream)

### PR DESCRIPTION
**Pull request by hinohie**
This patch will make some simple way to upload image from C# Stream to View.

- EncodedImageBuffer will create by C# Stream.
  EncodedImageBuffer buffer = new EncodedImageBuffer(stream);
- Generate ImageUrl
  ImageUrl url = buffer.GenerateUrl();
- You can use this ImageUrl as image's url
  ImageView view = new ImageView(url.GetUrl());

This patch required below dali patches
https://review.tizen.org/gerrit/#/c/platform/core/uifw/dali-csharp-binder/+/261019/
https://review.tizen.org/gerrit/#/c/platform/core/uifw/dali-toolkit/+/259415/

### CODE REVIEW

File: src/Tizen.NUI/src/public/Images/EncodedImageBuffer.cs [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        /// Get VectorUnsignedChar from System.IO.Stream.
+        /// This funcion is private (exist only for Constructor.)
+        /// </summary>
+        private static VectorUnsignedChar GetRawBuffrFromStreamHelper(System.IO.Stream stream)
+        {
+            if(stream == null)
+            {
+                throw new ArgumentNullException(nameof(stream));
+            }
+            long streamLength = stream.Length - stream.Position;
```
Comment (myroot) [inline src/Tizen.NUI/src/public/Images/EncodedImageBuffer.cs:128]: Do not use `Position` property, Only specific streams support it
https://docs.microsoft.com/en-us/dotnet/api/system.io.stream.position?view=net-5.0#remarks

Comment (hinohie) [reply src/Tizen.NUI/src/public/Images/EncodedImageBuffer.cs]: Fix done. Thank you for catch my mistake.
I make same mistake at `ImageLoading.cs`. I'll fix it another PR

File: src/Tizen.NUI/src/public/Images/EncodedImageBuffer.cs [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        {
+            if(stream == null)
+            {
+                throw new ArgumentNullException(nameof(stream));
+            }
+            if(!stream.CanRead)
+            {
+                throw new InvalidOperationException("stream don't support to read");
+            }
+            long streamLength = stream.Length;
```
Comment (myroot) [inline src/Tizen.NUI/src/public/Images/EncodedImageBuffer.cs:132]: `Length` also not support all types of Stream
https://docs.microsoft.com/ko-kr/dotnet/api/system.io.stream.length?view=net-5.0#--

You do better to copy a stream on MemoryStream

https://github.com/Samsung/TizenFX/blob/90572287f263121d0e1bceaeb1273f99e8d5cd24/src/ElmSharp/ElmSharp/Image.cs#L484-L501

Comment (hinohie) [reply src/Tizen.NUI/src/public/Images/EncodedImageBuffer.cs:132]: If we change like that, Whole stream reading is the only possible works. Is it okay?

File: src/Tizen.NUI/src/public/Images/EncodedImageBuffer.cs [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+            // Allocate buffer that internal DALi engine can read
+            VectorUnsignedChar buffer = new VectorUnsignedChar();
+
+            buffer.Resize((uint)readableStreamLength);
+            var bufferBegin = buffer.Begin();
+            global::System.Runtime.InteropServices.HandleRef bufferRef = SWIGTYPE_p_unsigned_char.getCPtr(bufferBegin);
+
+            // Copy data from memoryStream to buffer
+            System.Runtime.InteropServices.Marshal.Copy(streamData, 0, bufferRef.Handle, readableStreamLength);
```
Comment (myroot) [inline src/Tizen.NUI/src/public/Images/EncodedImageBuffer.cs:160]: How about use `GetBuffer` instead copy into `streamData`,
https://docs.microsoft.com/ko-kr/dotnet/api/system.io.memorystream.getbuffer?view=net-5.0
```suggestion

            // Allocate buffer that internal DALi engine can read
            VectorUnsignedChar buffer = new VectorUnsignedChar();

            buffer.Resize((uint)streamLength);
            var bufferBegin = buffer.Begin();
            global::System.Runtime.InteropServices.HandleRef bufferRef = SWIGTYPE_p_unsigned_char.getCPtr(bufferBegin);

            // Copy data from memoryStream to buffer
            System.Runtime.InteropServices.Marshal.Copy(memoryStream.GetBuffer(), 0, bufferRef.Handle, streamLength);
```

File: src/Tizen.NUI/src/public/Images/EncodedImageBuffer.cs (hunk 2 above)
Comment (myroot) [reply src/Tizen.NUI/src/public/Images/EncodedImageBuffer.cs]: I don't know exactly means of `Whole stream reading`. 
As I understand you need to know a stream size before create a native buffer.
So reading a whole stream is necessary to know a stream size.

Comment (myroot) [reply src/Tizen.NUI/src/public/Images/EncodedImageBuffer.cs]: If we possible to append data into native buffer without a final size, we can use Stream without copy to memoryStream

File: src/Tizen.NUI/src/public/Images/EncodedImageBuffer.cs (hunk 3 above)
Comment (hinohie) [reply src/Tizen.NUI/src/public/Images/EncodedImageBuffer.cs]: Good points. Fix done.

File: src/Tizen.NUI/src/public/Images/EncodedImageBuffer.cs (hunk 2 above)
Comment (hinohie) [reply src/Tizen.NUI/src/public/Images/EncodedImageBuffer.cs]: `VectorUnsignedChar` support `append` operation only for single byte.
(Appending raw-buffers are not support, but appending `VectorUnsignedChar` is support.)
That's why I get final size and copy it ;-;

Comment (hinohie) [reply src/Tizen.NUI/src/public/Images/EncodedImageBuffer.cs]: I imagine that some user push `used middle of stream` and try to load images from that.
But... I'll not care about them anymore.

**Review by myroot — APPROVED** [review]

**Review by myroot — APPROVED** [review]

**Pull request merged**